Library/Homebrew/test/formulary_spec.rb

```diff
@@ -370,8 +371,38 @@ def formula_json_contents(extra_items = {})
         }
       end
 
+      let(:blank_fields) do
+        {
+          "oldnames"                 => [],
+          "aliases"                  => [],
+          "versioned_formulae"       => [],
+          "keg_only_reason"          => nil,
+          "options"                  => [],
+          "build_dependencies"       => [],
+          "dependencies"             => [],
+          "test_dependencies"        => [],
+          "recommended_dependencies" => [],
+          "optional_dependencies"    => [],
+          "uses_from_macos"          => [],
+          "uses_from_macos_bounds"   => [],
+          "requirements"             => [],
+          "conflicts_with"           => [],
+          "conflicts_with_reasons"   => [],
+          "link_overwrite"           => [],
+          "caveats"                  => nil,
+          "installed"                => [],
+          "linked_keg"               => nil,
+          "deprecation_date"         => [],
+          "deprecation_reason"       => [],
+          "disable_date"             => [],
+          "disable_reason"           => [],
+          "service"                  => nil,
+          "variations"               => {},
+        }
+      end
+
       before do
-        allow(described_class).to receive(:loader_for).and_return(described_class::FormulaAPILoader.new(formula_name))
+        allow(Homebrew::EnvConfig).to receive(:no_install_from_api?).and_return(false)
```

Comment:
By adding the `ruby_source_path` to the hash above and by mocking things so that it thinks it should use the API we no longer have to mock the `Formulary.loader_for` method. This means that the tests should better represent how the code is used in real life. It doesn't seem to make the tests noticeably slower either.

Reply:
Very nice 🎉